akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -0,0 +1,234 @@
+   * if the actor is not terminated within the `minBackoff` duration.
+   *
+   * In addition to the calculated exponential back-off an additional
+   * random delay based the given `randomFactor` is added, e.g. 0.2 adds up to 20%
+   * delay. The reason for adding a random delay is to avoid that all failing
+   * actors hit the backend resource at the same time.
+   *
+   * You can retrieve the current child `ActorRef` by sending `BackoffSupervisor.GetCurrentChild`
+   * message to this actor and it will reply with [[akka.pattern.BackoffSupervisor.CurrentChild]]
+   * containing the `ActorRef` of the current child, if any.
+   *
+   * The `BackoffSupervisor`delegates all messages from the child to the parent of the
+   * `BackoffSupervisor`, with the supervisor as sender.
+   *
+   * The `BackoffSupervisor` forwards all other messages to the child, if it is currently running.
+   *
+   * The child can stop itself and send a [[akka.actor.PoisonPill]] to the parent supervisor
+   * if it wants to do an intentional stop.
+   *
+   * Exceptions in the child are handled with the default supervisionStrategy, which can be changed by using
+   * [[BackoffOptions#withSupervisorStrategy]] or [[BackoffOptions#withDefaultStoppingStrategy]]. A
+   * `Restart` will perform a normal immediate restart of the child. A `Stop` will
+   * stop the child, but it will be started again after the back-off duration.
+   *
+   * @param childProps the [[akka.actor.Props]] of the child actor that
+   *   will be started and supervised
+   * @param childName name of the child actor
+   * @param minBackoff minimum (initial) duration until the child actor will
+   *   started again, if it is terminated
+   * @param maxBackoff the exponential back-off is capped to this duration
+   * @param randomFactor after calculation of the exponential back-off an additional
+   *   random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
+   *   In order to skip this additional delay pass in `0`.
+   */
+  def onStop(
+    childProps: Props,
+    childName: String,
+    minBackoff: FiniteDuration,
+    maxBackoff: FiniteDuration,
+    randomFactor: Double): BackoffOptions =
+    BackoffOptionsImpl(StopImpliesFailure, childProps, childName, minBackoff, maxBackoff, randomFactor)
+}
+
+/**
+ * Configures a back-off supervisor actor. Start with `BackoffOptions.onStop` or `BackoffOptions.onFailure`.
+ * BackoffOptions is immutable, so be sure to chain methods like:
+ * {{{
+ * val options = BackoffOptions.onFailure(childProps, childName, minBackoff, maxBackoff, randomFactor)
+ *               .withManualReset
+ * context.actorOf(BackoffSupervisor.props(options), name)
+ * }}}
+ * or capture the result of a method in a variable:
+ * {{{
+ * var options = BackoffOptions.onFailure(childProps, childName, minBackoff, maxBackoff, randomFactor)
+ * options = options.withManualReset
+ * context.actorOf(BackoffSupervisor.props(options), name)
+ * }}}
```

Comment:
let's not show the `var` style :) People can figure it out if they need to, it's not something we want to encourage though I think